api/envoy/data/extensions/filter/udp/dns_filter/v2alpha/inline_dns_table.proto

```diff
@@ -0,0 +1,58 @@
+syntax = "proto3";
+
+package envoy.data.extensions.filter.udp.dns_filter.v2alpha;
+
+import "google/protobuf/duration.proto";
+
+import "udpa/annotations/status.proto";
+
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.data.extensions.filter.udp.dns_filter.v2alpha";
+option java_outer_classname = "InlineDnsTableProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).work_in_progress = true;
+
+// [#protodoc-title: DNS Filter Table Data]
+// :ref:`DNS Filter config overview <config_udp_listener_filters_dns_filter>`.
+
+// This message contains the configuration for the DNS Filter if populated
+// from the control plane
+message InlineDnsTable {
+  // This message contains a list of IP addresses returned for a query for a known name
+  message AddressList {
+    repeated string address = 1;
+  }
+
+  // This message type is extensible and can contain a list of addresses
+  // or dictate some other method for resolving the addresses for an
+  // endpoint
+  message DnsEndpoint {
+    oneof endpoint_config {
+      AddressList address_list = 1;
+    }
+  }
+
+  message DnsVirtualDomain {
+    // The domain name for which Envoy will respond to query requests
+    string name = 1 [(validate.rules).string = {min_len: 2 well_known_regex: HTTP_HEADER_NAME}];
```

Comment:
Should this potentially be repeated and/or support wildcards? This would avoid address duplication below?

Reply:
I wanted to add another layer of message so that we could extend this type and not the one containing this message.

We could condense the `DnsVirtualDomain` and use a `oneof`.   That sounds like it would condense this and achieve the same intent